### ISSUE TITLE
Service Principal auxiliary tenant

### ISSUE THREAD

**Issue by elsesiy** [Azure/aad-pod-identity#814]
The `AzureIdentitySpec` limits Service Principal usage to a given tenant denoted by the "tenantId" stanza. There are use cases in which the returned SP has been configured for accessing resources across tenants.

A token returned from the IMDS fails to access the remote tenant with a `401 InvalidAuthenticationTokenTenant` since the token issued is not aware of the auxiliary tenant.

Are there any plans to support such use cases?

Thanks!

**Comment by chewong** [comment]
Thanks for the suggestion. It's definitely a valid use case and I've put this issue in the backlog for now.

**Comment by chewong** [comment]
@elsesiy we just added this issue to our v1.7.0 milestone. ETA for the release is mid November.

**Comment by aramase** [comment]
@elsesiy We've published `v1.7.0-rc.0` which contains changes for multi tenant SP. 

```
mcr.microsoft.com/oss/azure/aad-pod-identity/mic:v1.7.0-rc.0
mcr.microsoft.com/oss/azure/aad-pod-identity/nmi:v1.7.0-rc.0
```

Example usage -

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: demo-aad1-sp
type: Opaque
data:
  clientSecret: CLIENT_SECRET
---
apiVersion: "aadpodidentity.k8s.io/v1"
kind: AzureIdentity
metadata:
  name: demo-aad1
spec:
  type: 1
  tenantID: TENANT_ID
  clientID: CLIENT_ID
  clientPassword: {"name":"demo-aad1-sp","namespace":"default"}
  auxiliaryTenantIDs:
  - <auxiliary tenantid 1>
  - <auxiliary tenantid 2>
```

NMI returns a slice of tokens. The first token is for the primary tenant ID as defined in the `spec.tenantID`. Tokens from [1:] are the tokens for auxiliary tenant ids. 

Try it out and let us know you're feedback.

**Comment by elsesiy** [comment]
Thanks @aramase, I'll take a look at this as part of the project I'm working on. Will provide feedback in the next couple of days!
